manim/utils/bezier.py

```diff
@@ -369,3 +370,108 @@ def get_quadratic_approximation_of_cubic(a0, h0, h1, a1):
+
+    Returns
+    -------
+        list[float]
+            List containing possible parameters for the given point on
+            the given bezier curve.
+            This usually only contains one or zero elements, but if the
+            point is, say, at the beginning/end of a closed loop, may return
+            a list with more than 1 value, corresponding to the beginning and
+            end etc. of the loop.
+
+    Raises
+    ------
+    :class:`ValueError`
+        When ``point`` and the control points have different shapes.
+    """
+    if not all(np.shape(point) == np.shape(c_p) for c_p in control_points):
+        raise ValueError(
+            f"Point {point} and Control Points {control_points} have different shapes."
+        )
+    control_points = np.array(control_points)
+    n = len(control_points) - 1
+
+    roots = []
+    for dim, coord in enumerate(point):
+        control_coords = c_c = control_points[:, dim]
+        terms = []
+        for term_power in range(n, -1, -1):
+            outercoeff = choose(n, term_power)
+            term = []
+            negative = 1
+            for subterm_num in range(term_power, -1, -1):
+                innercoeff = choose(term_power, subterm_num) * negative
+                subterm = (
+                    innercoeff * c_c[subterm_num]
+                    if term_power > 0
+                    else ((innercoeff * c_c[subterm_num]) - coord)
+                )
+                term.append(subterm)
+                negative = -1 if negative > 0 else 1
+            terms.append(outercoeff * sum(np.array(term)))
+        bezier_polynom = np.polynomial.Polynomial(terms[::-1])
+        polynom_roots = bezier_polynom.roots()
+        if len(polynom_roots) > 0:
+            roots.append(polynom_roots)
+
+    roots = np.around(roots, int(np.log10(1 / round_to)))
+    roots = [[root for root in rootlist if root.imag == 0] for rootlist in roots]
+    roots = reduce(np.intersect1d, roots)  # Get common roots.
+    roots = np.array([r.real for r in roots]).tolist()
+    return roots
+
+
+def point_lies_on_bezier(
+    point: typing.Iterable[typing.Union[float, int]],
+    control_points: typing.Iterable[typing.Iterable[typing.Union[float, int]]],
+    round_to: typing.Optional[typing.Union[float, int]] = 1e-6,
+) -> bool:
+    """Checks if a given point lies on the bezier curves with the given control points.
+
+    Parameters
+    ----------
+    point
+        The Cartesian Coordinates of the point to check.
+    control_points
+        The Cartesian Coordinates of the ordered control
+        points of the bezier curve on which the point may
+        or may not lie.
+    round_to
+        A float whose number of decimal places all values
+        such as coordinates of points will be rounded.
+
+    Returns
+    -------
+    bool
+        Whether the point lies on the curve.
+    """
+    # Method taken from
+    # http://polymathprogrammer.com/2012/04/03/does-point-lie-on-bezier-curve/
```

Comment:
I think the comment is better suited to be here because the more complicated call is mostly just building the bezier polynomial, which isn't what the article details.

Reply:
as far as I can tell, I still think it should be in the other call, at least how "bezier_params_from_point" is named.  You're getting the t values of a point, and if there are any t-values that work, then the point is on the curve - that's how I'm parsing that line of code. None of that is complicated and needs a link (though perhaps it needs slightly better named variables, but not a blocking issue). Rather, the incantation in bezier_prams_from_point does not make sense to me until I read linked article. At the very least, it is worth noting that the strategy is solving cubic equations of x and y for t.

Reply:
I see your point. It's better to have the "explanation" in the more complicated call - even if it's not the main focus of the article. 
I guess it wouldn't hurt to add a couple comments on how the function works as well.